lib/adagrams.rb

```diff
@@ -0,0 +1,218 @@
+    words_created = []
+
+    input_array = []
+    input.length.times do |i|
+      input_array << input[i]
+    end
+
+    testing_array = []
+    editted_letters_drawn = letters_drawn_input
+
+    input_array.length.times do |i|
+      if editted_letters_drawn.include?(input_array[i])
+        index_of_letter = editted_letters_drawn.index(input_array[i])
+        editted_letters_drawn.delete_at(index_of_letter)
+
+        testing_array << true
+
+      else
+
+        testing_array << false
+      end
+
+    end
+
+
+
+    if testing_array.all?
+      # @words_created << input
+      return true
+    else
+      return false
+    end
+
+
+  end
+
+
+  def score_word(word)
+
+    parsed_word = []
+    word.length.times do |i|
+      parsed_word << word[i].upcase
+    end
+
+    total = 0
+    parsed_word.each do |letter|
+      case letter
+      when "A", "E", "I", "O", "U", "L", "N", "R", "S", "T"
+        value = 1
+        total += value
+      when "D", "G"
+        value = 2
+        total += value
+      when "B", "C", "M", "P"
+        value = 3
+        total += value
+      when "F", "H", "V", "W", "Y"
+        value = 4
+        total += value
+      when "K"
+        value = 5
+        total += value
+      when "J", "X"
+        value = 8
+        total += value
+      when "Q", "Z"
+        value = 10
+        total += value
+      end
+    end
+    if word.length >= 7
+      total += 8
+    end
+    if word.length == 0
+      total = 0
+    end
+    return total
+  end
+
+
+  def highest_score_from_words(words)
+
+    winning_words = []
+    length_of_winning_word = []
+    points_of_all_words_entered = []
+
+
+    indexes_of_max_point_words = []
+    final_winner_hash = {}
+
+    words.each do |x|
+      score = score_word(x)
+      points_of_all_words_entered << score
+    end
+
+    max_point_earned = points_of_all_words_entered.max
+
+    words.each do |x|
+      word_score = score_word(x)
+
+      if word_score == max_point_earned
+        indexes_of_max_point_words << words.find_index(x)
+      end
+
+    end
+
+    indexes_of_max_point_words.each do |y|
+      winning_words << words[y]
+    end
+
+    winning_words.each do |z|
+      length_of_winning_word << z.length
+    end
+
+    length_of_shortest_winning_word = length_of_winning_word.min
+    all_winning_words_of_shortest_length = []
+    number_of_winning_words_of_min_length = 0
+
+    winning_words.each do |x|
+      if x.length == length_of_shortest_winning_word
+        number_of_winning_words_of_min_length += 1
+        all_winning_words_of_shortest_length << x
+      end
+    end
+
+
+    length_of_winning_word.length.times do |i|
+      if length_of_winning_word[i] == 10
+
+        final_winner_hash[:word] = winning_words[i]
+        final_winner_hash[:score] = score_word(winning_words[i])
+        break
+      else
+        if number_of_winning_words_of_min_length == 1
+
+          final_winner_hash[:word] = all_winning_words_of_shortest_length
```

Comment:
Here, you are assigning into `final_winner_hash[:word]` the value of the entire array `all_winning_words_of_shortest_length`... even if this array is one element long, you'll still want to take it out of this array with `all_winning_words_of_shortest_length.first` or `all_winning_words_of_shortest_length[0]`.